central/complianceoperator/v2/report/manager/manager_impl.go

```diff
@@ -0,0 +1,95 @@
+package manager
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/pkg/errors"
+	scanConfigurationDS "github.com/stackrox/rox/central/complianceoperator/v2/scanconfigurations/datastore"
+	"github.com/stackrox/rox/generated/storage"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/sync"
+)
+
+var (
+	log         logging.Logger
+	maxRequests = 10
+)
+
+type reportRequest struct {
+	scanConfig *storage.ComplianceOperatorScanConfigurationV2
+}
+
+type managerImpl struct {
+	datastore            scanConfigurationDS.DataStore
+	runningReportConfigs map[string]*reportRequest
+	//channel for report job requests
+	reportRequests chan *reportRequest
+	stopper        concurrency.Stopper
+
+	//Mutex to synchronize access to runningReportConfigs map
+	mu sync.Mutex
+}
+
+func New(scanConfigDS scanConfigurationDS.DataStore) Manager {
+	return &managerImpl{
+		datastore:            scanConfigDS,
+		stopper:              concurrency.NewStopper(),
+		runningReportConfigs: make(map[string]*reportRequest, maxRequests),
+		reportRequests:       make(chan *reportRequest, maxRequests),
+	}
+}
+
+func (m *managerImpl) SubmitReportRequest(ctx context.Context, scanConfig *storage.ComplianceOperatorScanConfigurationV2) error {
```

Comment:
@ksurabhi91 Do we need the whole config or just the ID and/or name?

Reply:
We want to store scanconfig in the in memory struct. so API layer will pass while config to SubmitReportRequest

Reply:
Right but are we planning to use the whole object for anything?  I'm thinking I would probably just pass the ID or name for now.  If the manager needs the details, it can look them up.  IMO.

Reply:
so I think the idea is to read config in service layer and validate,persist the  report generation request. Then send scan config details to the manager. Since this is v1 we are sending the scan config as it is and dont do any validation. 
This is how vuln reporting works as well

Reply:
Long term, yeah that makes sense.  So it is ok to pass the object to the manager.  I'm not going to nitpick that.